packages/flutter_web_plugins/test/history/url_strategy_test.dart

```diff
@@ -0,0 +1,153 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:html';
+
+@TestOn('chrome') // Uses web-only Flutter SDK
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_web_plugins/flutter_web_plugins.dart';
+
+void main() {
```

Comment:
What about `pushState` and others?

Reply:
Handling the path is the only tricky part. Other methods are simply calling their PlatformLocation counterparts.